### DISCUSSION TITLE
Can Streamlink accept a piped stream from another program as input?

### DISCUSSION THREAD

**Discussion by zz478** [streamlink/streamlink#6326] [Q&A] [answered]
### Description

I was told that Streamlink might help with an issue I have but after reading the documentation I am not so sure.  I have a program that will send an audio/video transport stream out as standard output or on a named pipe.  I have another program that will take that stream as input and save it to a storage device.  The problem is the output of the first program sometimes glitches or is momentarily interrupted or perhaps even momentarily changes format, honestly I am not completely certain what happens but whatever it is, it is enough to cause the second program to think the stream is finished and stop recording.

I understand that Streamlink has some way to deal with this, such as when ads are sent out in a different format from the main program.  But as far as I can tell, the only option for Streamlink to accept input is from one of its plugins, and also as far as I can tell there is no plugin that will accept input from standard input or a named pipe.  I just need to be able to pipe the output of program 1 into something (Streamlink?) that will that that input and normalize it, so to speak, and then send the normalized output to program 2 via a second pipe.  So like:

program1 (options) | streamlink (options) | program2 (options).

Am I correct in thinking there is no way to do this using Streamlink, or am I missing something?  If it's in the documentation I sure did not see it, but please feel free to point me to the correct section if it is really in there. Thank you.

**Comment by bastimeyer** [comment, marked as answer]
I think you completely miss the point of Streamlink and its functionality.

While you can make Streamlink read from stdin using `httpstream://file:///dev/stdin best`, it'd merely act as another buffer, same as `cat`. There is nothing like what you call "normalization". An interruption of the input stream or a stream discontinuity where the bitstream is not coherent anymore will be the same on the other end of the pipe.

If you want to re-mux or re-encode video/audio input, then use FFmpeg.
